### ISSUE TITLE
libressl fails on `make -j4 install`

### ISSUE THREAD

**Issue by chipitsine** [libressl-portable/portable#461]
tried under travis-ci when building https://github.com/SoftEtherVPN/SoftEtherVPN

`make install` = ok
`make -j4 install` = fail

**Comment by busterb** [comment]
I can understand `make -j4` but why parallelize `make -j4 install`? It maybe saves 1/10th of a second:

```
$ time sudo make install > /dev/null
libtool: warning: relinking 'libssl.la'
libtool: warning: relinking 'libtls.la'

real	0m3.661s
user	0m2.401s
sys	0m0.483s
$ time sudo make -j4 install > /dev/null
libtool: warning: relinking 'libssl.la'
libtool: warning: relinking 'libtls.la'

real	0m3.537s
user	0m2.358s
sys	0m0.524s
```

**Comment by chipitsine** [comment]
it was  `make all install` which I tried to speed up as `make -j4 all install`

**Comment by busterb** [comment]
Do `make -j4 && make install`

**Comment by chipitsine** [comment]
I did that already.

however, failing to `make -j4 all install` looks like a bug, so I decided to open an issue

**Comment by busterb** [comment]
What error did you get? Even running the install / build in a single step works fine for me.

**Comment by chipitsine** [comment]
I will provide debug tomorrow

**Comment by busterb** [comment]
Thanks!

**Comment by busterb** [comment]
I tried to find a failure in the travis logs for SoftEtherVPN but could not find one. Nice build automation, by the way!

**Comment by chipitsine** [comment]
I did it before merge.

wait a moment, I'll provide links

**Comment by chipitsine** [comment]
two builds

https://travis-ci.org/chipitsine/SoftEtherVPN/jobs/454948526
https://travis-ci.org/chipitsine/SoftEtherVPN/jobs/454951657


as for build automation, if you are interested, I can do some automation for libressl-portable

**Comment by busterb** [comment]
Thanks, these logs look like it failed between building the static and the dynamic library simultaneously, the object cleanup from the first library build pass probably killed the wrong file during compilation of the second build pass:

```
  CCLD     libcompat.la
  CCLD     libcompatnoopt.la
  CCLD     libcrypto.la
  CC       x509v3/libcrypto_la-v3_akey.lo
  CC       x509v3/libcrypto_la-v3_akeya.lo
gcc: error: x509v3/.libs/libcrypto_la-v3_akeya.o: No such file or directory
make[2]: *** [libcrypto.la] Error 1
make[2]: Leaving directory `/home/travis/build/chipitsine/SoftEtherVPN/libressl-2.8.2/crypto'
make[1]: *** [all] Error 2
```

I don't think this is in code we actually can control, since all of the dependencies appear to be properly specified in Makefile.am. Whether shared and static library build rules conflict with each is going to be up to the interaction of GNU make and/or autoconf/automake's output in combination with the tools that are part of Ubuntu 14.04, and the options you specify.

A possibility could be disabling parallel builds entirely, which I'm not keen to do to support this use-case, especially since there is a simple workaround: build, then install, as 2 steps. And with this, you get the added bonus of building as a non-root user, and only installing as a privileged user. If that is not attractive (e.g. you're just doing an embedded install), you could also specify '--disable-shared' or whatever is required to just build the library variant your project needs, since it seems unlikely an embedded instance needs both shared and static libraries.

TL;DR: I don't see a bug in the build files for this project, or an attractive workaround that I'd be willing to integrate at the moment that doesn't kill parallel jobs altogether. Feel free to reopen this ticket if you do see something we can add to keep automake from stepping on its toes for parallel shared/static library builds though.

**Comment by chipitsine** [comment]
I did not check higher version of autoconf/automake
if I will ever have time, I'll check, maybe we will end up with "we need autoconf N.N.N or higher"

thank for the review!
